src/Serilog/Context/LogContext.cs

```diff
@@ -75,37 +76,80 @@ public static class LogContext
         /// <returns>A token that must be disposed, in order, to pop properties back off the stack.</returns>
         public static IDisposable PushProperty(string name, object value, bool destructureObjects = false)
         {
+            return Push(new PropertyEnricher(name, value, destructureObjects));
+        }
+
+        /// <summary>
+        /// Push an enricher onto context, returning an <see cref="IDisposable"/>
+        /// that must later be used to remove the property, along with any others that
+        /// may have been pushed on top of it and not yet popped. The property must
+        /// be popped from the same thread/logical call context.
+        /// </summary>
+        /// <param name="enricher">Log Properties to push onto the log context</param>
+        /// <returns>A token that must be disposed, in order, to pop properties back off the stack.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static IDisposable Push(ILogEventEnricher enricher)
+        {
+            if (enricher == null) throw new ArgumentNullException(nameof(enricher));
+
             var stack = GetOrCreateEnricherStack();
             var bookmark = new ContextStackBookmark(stack);
 
-            Enrichers = stack.Push(new PropertyEnricher(name, value, destructureObjects));
+            Enrichers = stack.Push(enricher);
 
             return bookmark;
         }
 
         /// <summary>
-        /// Push multiple properties onto the context, returning an <see cref="IDisposable"/>
-        /// that can later be used to remove the properties. The properties must
+        /// Push multiple enrichers onto the context, returning an <see cref="IDisposable"/>
+        /// that must later be used to remove the property, along with any others that
+        /// may have been pushed on top of it and not yet popped. The property must
```

Comment:
The context is a stack, but each "push" produces the `IDisposable` bookmark that will pop the stack all the way to whatever state it was in prior to the Push(). This means:

```
x = Push();
Push();
Push();
x.Dispose();
```

will pop all three enrichers off the stack.